pkg/auth/providers/azure/azure_provider.go

```diff
@@ -211,61 +202,27 @@ func (ap *azureProvider) TransformToAuthProvider(
 	return p, nil
 }
 
-func (ap *azureProvider) loginUser(
-	azureCredential *v32.AzureADLogin,
-	config *v32.AzureADConfig,
-	test bool,
-) (v3.Principal, []v3.Principal, string, error) {
+func (ap *azureProvider) loginUser(azureCredential *v32.AzureADLogin, config *v32.AzureADConfig, test bool) (v3.Principal, []v3.Principal, string, error) {
 	var err error
-
 	if config == nil {
 		config, err = ap.getAzureConfigK8s()
 		if err != nil {
 			return v3.Principal{}, nil, "", err
 		}
 	}
 
-	logrus.Debug("[AZURE_PROVIDER] Started token swap with AzureAD")
-
-	azureClient, err := newClientCode(azureCredential.Code, config)
-	if err != nil {
-		return v3.Principal{}, nil, "", err
-	}
-
-	logrus.Debug("[AZURE_PROVIDER] Completed token swap with AzureAD")
-
-	oid, err := parseJWTforField(azureClient.accessToken(), "oid")
-	if err != nil {
-		return v3.Principal{}, nil, "", err
-	}
-
-	logrus.Debug("[AZURE_PROVIDER] Started getting user info from AzureAD")
-
-	user, err := azureClient.userClient.Get(context.Background(), oid)
-	if err != nil {
-		return v3.Principal{}, nil, "", err
-	}
-
-	logrus.Debug("[AZURE_PROVIDER] Completed getting user info from AzureAD")
-
-	userPrincipal := ap.userToPrincipal(user)
-	userPrincipal.Me = true
-
-	var mem bool
-	params := graphrbac.UserGetMemberGroupsParameters{
-		SecurityEnabledOnly: &mem,
+	var useMSGraph bool
+	if !graphEndpointDeprecated(config) {
+		useMSGraph = true
 	}
-
-	userGroups, err := azureClient.userClient.GetMemberGroups(context.Background(), *user.ObjectID, params)
+	azureAuthenticator := clients.NewAuthenticator(useMSGraph)
```

Comment:
This is mostly related to my questions that I have posed in other places, but I think it's worth restating here:

1) Use of "Authenticator" here is confusing - is is a client same as elsewhere.
2) Why don't we initialize the client using the NewClient method that we have in the provider? It would prevent us from doing the Re-init client calls that we do in the client's login user methods.